### PR TITLE
Fix freeze on long indexation for snapping on intersections

### PR DESCRIPTION
**Pull request by Djedouas**
If snap on intersections is enabled, this specific edge search with the locator was not in relaxed mode, whereas the standard snapping is happening in relaxed mode.

As a result, trying to use the snapping during the first indexation was freezing QGIS while the indexation is happening, waiting for it to end.

On a layer where the indexation is longer than the timeout (30sec, i.e. a WFS layer as in issue #51179), the locator stops abruptly and resets itself, crashing the indexation and QGIS.

I propose to use the same match options for the edge search as for the best match search. Along with some formatting.

The tests that had the intersection snapping activated have been adapted accordingly, due to the fact that the all the layer indexations that were blocking. Now we must wait for it to end.

### CODE REVIEW

File: tests/src/app/maptooladdfeatureline/testqgsmaptooladdfeatureline.cpp [hunk 1]
```diff
@@ -1006,6 +1006,11 @@ void TestQgsMapToolAddFeatureLine::testWithTopologicalEditingDifferentCanvasCrs(
   snapConfig.project()->setTopologicalEditing( true );
   mCanvas->snappingUtils()->setConfig( snapConfig );
 
+  // Wait for indexing to complete
+  if ( auto loc = mCanvas->snappingUtils()->locatorForLayer( mLayerCRS3946Line ) )
+    if ( loc->isIndexing() )
+      loc->waitForIndexingFinished();
```
Comment (nyalldawson) [inline tests/src/app/maptooladdfeatureline/testqgsmaptooladdfeatureline.cpp:1012]:
```suggestion
  if ( QgsPointLocator * loc = mCanvas->snappingUtils()->locatorForLayer( mLayerCRS3946Line ) )
  {
    if ( loc->isIndexing() )
    {
      loc->waitForIndexingFinished();
    }
  }
```

File: tests/src/app/maptooladdfeatureline/testqgsmaptooladdfeatureline.cpp [hunk 2]
```diff
@@ -1086,6 +1091,11 @@ void TestQgsMapToolAddFeatureLine::testWithTopologicalEditingWIthDiffLayerWithDi
   snapConfig.project()->setTopologicalEditing( true );
   mCanvas->snappingUtils()->setConfig( snapConfig );
 
+  // Wait for indexing to complete
+  if ( auto loc = mCanvas->snappingUtils()->locatorForLayer( mLayerCRS3945Line ) )
+    if ( loc->isIndexing() )
+      loc->waitForIndexingFinished();
```
Comment (nyalldawson) [inline tests/src/app/maptooladdfeatureline/testqgsmaptooladdfeatureline.cpp:1097]:
```suggestion
  if ( QgsPointLocator * loc = mCanvas->snappingUtils()->locatorForLayer( mLayerCRS3945Line ) )
  {
    if ( loc->isIndexing() )
    {
      loc->waitForIndexingFinished();
    }
  }
```

Comment (nyalldawson) [reply tests/src/app/maptooladdfeatureline/testqgsmaptooladdfeatureline.cpp:1097]: (elsewhere too)

File: src/core/qgssnappingutils.cpp [hunk 3]
```diff
@@ -290,21 +290,15 @@ QgsPointLocator::Match QgsSnappingUtils::snapToMap( const QgsPointXY &pointMap,
...
-    }
+      edges = loc->edgesInRect( pointMap, tolerance, filter, relaxed );
 
     for ( QgsVectorLayer *vl : mExtraSnapLayers )
     {
       QgsPointLocator *loc = locatorForLayerUsingStrategy( vl, pointMap, tolerance );
       _updateBestMatch( bestMatch, pointMap, loc, type, tolerance, filter, false );
       if ( mSnappingConfig.intersectionSnapping() )
-        edges << loc->edgesInRect( pointMap, tolerance );
+        edges << loc->edgesInRect( pointMap, tolerance, filter, false );
```
[the remark this answers is not in the document]
Comment (Djedouas) [reply src/core/qgssnappingutils.cpp:301]: relaxed is hard-coded to false when considering extra snap layers (on `_updateBestMatch`), so I followed the same logic

**Comment by Djedouas** [comment]
@troopa81 @nyalldawson thanks for the review, I pushed the suggestions

**Review by troopa81 — APPROVED** [review]

**Pull request merged**